### ISSUE TITLE
/usr/bin/env: ‘bash\r’: No such file or directory when running flutter doctor

### ISSUE THREAD

**Issue by a-sajjad72** [flutter/website#8615]
### Describe the problem

trying to build the site in windows 10 on cygwin terminal (bash shell).
shows error at  
`ERROR [flutter 5/7] RUN flutter doctor`

### Additional context

```
$ make setup
make clean
make[1]: Entering directory `E:/LECTURES/Development/Documentation/flutter_docs'
rm -rf _site .jekyll* src/.jekyll* *.log tmp .dart_tool
make[1]: Leaving directory `E:/LECTURES/Development/Documentation/flutter_docs'
docker-compose build --no-cache site
[+] Building 863.2s (12/24)
 => [internal] load build definition from Dockerfile                       1.7s
 => => transferring dockerfile: 3.85kB                                     0.1s
 => [internal] load .dockerignore                                          2.1s
 => => transferring context: 146B                                          0.0s
 => [internal] load metadata for docker.io/library/ruby:3.2-2-slim-bulls  11.6s
 => [internal] load build context                                        497.0s
 => => transferring context: 1.39GB                                      492.1s
 => CACHED [base 1/3] FROM docker.io/library/ruby:3.2-2-slim-bullseye@sha  0.0s
 => [base 2/3] RUN apt-get update && apt-get install -yq --no-install-r  820.8s
 => [base 3/3] WORKDIR /app                                                3.6s
 => [flutter 1/7] COPY ./flutter ./flutter                                 7.3s
 => [flutter 2/7] COPY ./site-shared ./site-shared                         2.4s
 => [flutter 3/7] COPY pubspec.yaml ./                                     1.8s
 => [flutter 4/7] RUN if test -n "" -a "" != "stable" ; then       cd flu  7.4s
 => ERROR [flutter 5/7] RUN flutter doctor                                 4.3s
------
 > [flutter 5/7] RUN flutter doctor:
#0 3.472 /usr/bin/env: ‘bash\r’: No such file or directory
------
failed to solve: executor failed running [/bin/sh -c flutter doctor]: exit code:
 127
make: *** [setup] Error 17
```

**Comment by parlough** [comment]
Thanks for opening the issue!

It seems the issue might be with Window's CRLF line endings. I'm not sure if this will work, but can you try cloning the repository with the `--config core.autocrlf=input` flag and trying again?

```bash
git clone --recurse-submodules https://github.com/flutter/website.git --config core.autocrlf=input
```

While I haven't tried it in a while, you might find more success storing and running the site completely in a [WSL2 instance](https://learn.microsoft.com/en-us/windows/wsl/install).

**Comment by parlough** [comment]
You ran that command within cygwin? Did you get the same error as before? 

I don't have the ability to test a Windows setup right now, but I think your best bet will be cloning and running within a [WSL instance](https://learn.microsoft.com/en-us/windows/wsl/install).

**Comment by a-sajjad72** [comment]
> You ran that command within cygwin? Did you get the same error as before?

@parlough yeah i ran that command in cygwin and got the same error as before.

> I don't have the ability to test a Windows setup right now, but I think your best bet will be cloning and running within a [WSL instance](https://learn.microsoft.com/en-us/windows/wsl/install).

i cloned it with the normal command which is mentioned in the Readme [here](https://github.com/flutter/website#2-clone-this-repo-and-its-submodules) and build it using wsl, i got the same error as before.

but when cloned and build it in wsl the error was gone. successfully build the website.

but i have some questions regarding to access the build website.
1. do i have to start the docker container everytime whenever i want to access the docs.
2. is there any way to have these docs in html. just click the index.html file and have fun. i build the website to access it offline that's why i am asking.

**Comment by atsansone** [comment]
To be clear, this is not a Flutter issue, but a Windows Git issue.

Check out this [StackOverflow post](https://stackoverflow.com/questions/29045140/env-bash-r-no-such-file-or-directory) that explains the issue.

TL;DR: Git translated UNIX line breaks to Windows line breaks on `git clone`. `bash` has no idea what to do with those line breaks.

@skytomo221 : You were close. 

```terminal
$ git config --global core.autocrlf input
```

That ensured that all _writes_ to git and GitHub use LF only. 

```terminal
$ git config --global core.autocrlf false
```

This ensures that all _reads_ to git and GitHub use LF only. 

A deeper dive can be found on [this GitHub page](https://docs.github.com/en/get-started/getting-started-with-git/configuring-git-to-handle-line-endings).
